internal/provider/kubernetes/controller.go

```diff
@@ -1460,6 +1483,42 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 	return nil
 }
 
+func (r *gatewayAPIReconciler) processServiceCluster(ctx context.Context, resourceName string, resourceTree *resource.Resources, resourceMap *resourceMappings) error {
+	proxySvcName := proxy.ExpectedResourceHashedName(resourceName)
+	proxySvc := &corev1.Service{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: proxySvcName, Namespace: r.namespace}, proxySvc); err != nil {
+		return err
+	}
+	resourceTree.Services = append(resourceTree.Services, proxySvc)
+	resourceMap.allAssociatedBackendRefs.Insert(gwapiv1.BackendObjectReference{
+		Kind:      ptr.To(gwapiv1.Kind("Service")),
+		Namespace: gatewayapi.NamespacePtr(proxySvc.Namespace),
+		Name:      gwapiv1.ObjectName(proxySvc.Name),
+	})
+	resourceMap.allAssociatedNamespaces.Insert(proxySvc.Namespace)
+
+	proxyEndPtSlices := &discoveryv1.EndpointSliceList{}
```

Comment:
I dont think this is needed, its happening later in https://github.com/envoyproxy/gateway/blob/bebede52296bdce4bdd45df415a9da0b9bd84f98/internal/provider/kubernetes/controller.go#L577
I believe the only thing needed is to add the backendRef to the resourceMap

Reply:
You're right, fixed